### ISSUE TITLE
deployments.apps "cattle-cluster-agent" not found ... dial tcp 127.0.0.1:6080: connect: connection refused seen after provisioning cluster

### ISSUE THREAD

**Issue by sevko21** [rancher/rancher#29633]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Create a new Cluster with AWS EC2 Template. Rancheros is used as ami as described in the documentation
All ports were opened up during provisioning from rancher resulting in security group  rancher-nodes

**Result:**
After provisioning everything comes up. The health check results in:

Cluster health check failed: deployments.apps "cattle-cluster-agent" not found; Error while applying agent YAML, it will be retried automatically: exit status 1, Error from server (InternalError): an error on the server ("dial tcp 127.0.0.1:6080: connect: connection refused") has prevented the request from succeeding

**Environment information**
- Rancher version : 2.5.1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Infrastructure Provider AWS
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud
- Kubernetes version (use `kubectl version`):  1.19.3

- Docker version (use `docker version`):
19.3.11
```

**Comment by Yomanz** [comment]
I get this with RKE too. Rancher v2.5.1 HA

**Comment by superseb** [comment]
@sevko21 @Yomanz Please describe the exact steps to reproduce (what node driver, what node template config options used like image used and checked/unchecked options, how many node templates with what roles)

**Comment by Yomanz** [comment]
I just made a new cluster with "Existing Nodes" and it fully installs and then after it says `Waiting for the API to be available` it changes to the above error.

**Comment by superseb** [comment]
@Yomanz @sevko21 can you share `docker ps -a` from the node please?

**Comment by Yomanz** [comment]
https://ptero.co/raw/ovayuvorec
Sorry about the delay.

**Comment by sevko21** [comment]
Steps to reproduce: Created a node Template for AWS. ami-0b63175a9bbbcda85 for rancheros was chosen in the template (tried also with ubuntu). After creating a cluster with this template the nodes come up, the kubernetes cluster is created and at the point where the heathcheck is running the error message comes up

**Comment by sevko21** [comment]
Cluster health check failed: deployments.apps "cattle-cluster-agent" not found
Cluster health check failed: deployments.apps "cattle-cluster-agent" not found; Error while applying agent YAML, it will be retried automatically: exit status 1, Error from server (InternalError): an error on the server ("dial tcp 127.0.0.1:6080: connect: connection refused") has prevented the request from succeeding

CONTAINER ID        IMAGE                                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
fa824afb1f46        rancher/metrics-server                    "/metrics-server --k…"   27 minutes ago      Up 27 minutes                                   k8s_metrics-server_metrics-server-8449844bf-4648t_kube-system_d741f5c7-7798-4768-9118-358102d75587_0
ec662c47fc97        rancher/rancher-agent:v2.5.1              "run.sh --server htt…"   27 minutes ago      Up 27 minutes                                   optimistic_raman
4167dbae9dad        rancher/rancher-agent:v2.5.1              "run.sh -- share-roo…"   27 minutes ago      Exited (0) 27 minutes ago                       share-mnt
03d357cf07d3        rancher/cluster-proportional-autoscaler   "/cluster-proportion…"   27 minutes ago      Up 27 minutes                                   k8s_autoscaler_coredns-autoscaler-79599b9dc6-92nvr_kube-system_811ab367-46bc-4f44-a772-bbc5cad1edb7_0
d8346f6d7778        rancher/coreos-flannel                    "/opt/bin/flanneld -…"   27 minutes ago      Up 27 minutes                                   k8s_kube-flannel_canal-ctm94_kube-system_f1097d37-ba74-4ff7-aff5-fe7982954072_0
51712ed8728a        rancher/pause:3.2                         "/pause"                 27 minutes ago      Up 27 minutes                                   k8s_POD_coredns-autoscaler-79599b9dc6-92nvr_kube-system_811ab367-46bc-4f44-a772-bbc5cad1edb7_7
704ac10c78e6        rancher/pause:3.2                         "/pause"                 27 minutes ago      Up 27 minutes                                   k8s_POD_metrics-server-8449844bf-4648t_kube-system_d741f5c7-7798-4768-9118-358102d75587_7
72b1e246d771        rancher/calico-node                       "start_runit"            27 minutes ago      Up 27 minutes                                   k8s_calico-node_canal-ctm94_kube-system_f1097d37-ba74-4ff7-aff5-fe7982954072_0
147bfa1b1114        rancher/nginx-ingress-controller          "/usr/bin/dumb-init …"   27 minutes ago      Up 27 minutes                                   k8s_nginx-ingress-controller_nginx-ingress-controller-clzxv_ingress-nginx_91708d88-dd0f-4e93-81c7-0b6f91eb1e2e_0
7c4286390151        rancher/pause:3.2                         "/pause"                 27 minutes ago      Up 27 minutes                                   k8s_POD_nginx-ingress-controller-clzxv_ingress-nginx_91708d88-dd0f-4e93-81c7-0b6f91eb1e2e_0
e392ca1e2efd        rancher/calico-pod2daemon-flexvol         "/usr/local/bin/flex…"   27 minutes ago      Exited (0) 27 minutes ago                       k8s_flexvol-driver_canal-ctm94_kube-system_f1097d37-ba74-4ff7-aff5-fe7982954072_0
3e5e9ef846e5        rancher/calico-cni                        "/opt/cni/bin/install"   27 minutes ago      Exited (0) 27 minutes ago                       k8s_install-cni_canal-ctm94_kube-system_f1097d37-ba74-4ff7-aff5-fe7982954072_0
1167c926085c        rancher/pause:3.2                         "/pause"                 27 minutes ago      Up 27 minutes                                   k8s_POD_canal-ctm94_kube-system_f1097d37-ba74-4ff7-aff5-fe7982954072_0
30ed1b9d1db7        rancher/hyperkube:v1.19.3-rancher1        "/opt/rke-tools/entr…"   28 minutes ago      Up 27 minutes                                   kube-proxy
18a81bbc38b6        rancher/hyperkube:v1.19.3-rancher1        "/opt/rke-tools/entr…"   28 minutes ago      Up 28 minutes                                   kubelet
7d903407e454        rancher/hyperkube:v1.19.3-rancher1        "/opt/rke-tools/entr…"   28 minutes ago      Up 28 minutes                                   kube-scheduler
fd593985efb9        rancher/hyperkube:v1.19.3-rancher1        "/opt/rke-tools/entr…"   28 minutes ago      Up 28 minutes                                   kube-controller-manager
f3bb4e9a1562        rancher/hyperkube:v1.19.3-rancher1        "/opt/rke-tools/entr…"   28 minutes ago      Up 28 minutes                                   kube-apiserver
cd468a4d3db7        rancher/rke-tools:v0.1.65                 "/bin/bash"              28 minutes ago      Created                                         service-sidekick
067adfa1adb8        rancher/coreos-etcd:v3.4.13-rancher1      "/usr/local/bin/etcd…"   28 minutes ago      Up 28 minutes                                   etcd
cee39e92d691        rancher/rke-tools:v0.1.65                 "/docker-entrypoint.…"   30 minutes ago      Exited (0) 30 minutes ago                       cluster-state-deployer
40edc0d135ed        rancher/rancher-agent:v2.5.1              "run.sh --server htt…"   30 minutes ago      Up 30 minutes                                   relaxed_hertz

**Comment by sevko21** [comment]
looks like [https://github.com/rancher/rancher/issues/29239]

**Comment by superseb** [comment]
@sevko21 The error is not identical, are you running Rancher on the Rancher managed clusters?

**Comment by sevko21** [comment]
Rancher is running on a HA Cluster on three nodes onpremise. The mentioned aws cluster is provisoned with rancher new cluster option

**Comment by izaac** [comment]
@sevko21 I tried to reproduce this issue in `v2.5.1`

RancherOS AMI: ami-08faee1387289d316 (us-east-2) this was also taken from the documentation https://github.com/rancher/os/blob/master/README.md/#user-content-amazon

I created two clusters one with 1 etcd, 1 control plane, 3 workers
I used a custom Security group with all incoming and outgoing communication open.
The cluster got active.

The other cluster is 1 master (etcd/control plane), 1 worker.
I used the node template default of generating a rancher-nodes security group.
The cluster got active state

Sample node docker info

```
docker info
Client:
 Debug Mode: false

Server:
 Containers: 24
  Running: 20
  Paused: 0
  Stopped: 4
 Images: 14
 Server Version: 19.03.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.14.138-rancher
 Operating System: RancherOS v1.5.6
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 7.693GiB
 Name: iz-ec2-test2-wrkr-1
 ID: HTJA:TYOM:36N5:34KV:YSJ5:PTFH:W5IL:H32A:NJ3P:3PXM:6WZD:ILAK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
  provider=amazonec2
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

Sample `docker ps -a` 

```
 docker ps -a
CONTAINER ID        IMAGE                                             COMMAND                  CREATED              STATUS                     PORTS               NAMES
24ff7ac02012        e2987812f35f                                      "fleetagent"             About a minute ago   Up About a minute                              k8s_fleet-agent_fleet-agent-64d854c5b-ltsx4_fleet-system_a7a6c8d3-a87d-4367-9066-6902de9a7281_0
51bff26bf087        rancher/pause:3.2                                 "/pause"                 About a minute ago   Up About a minute                              k8s_POD_fleet-agent-64d854c5b-ltsx4_fleet-system_a7a6c8d3-a87d-4367-9066-6902de9a7281_0
bd45ab2daeb0        rancher/nginx-ingress-controller-defaultbackend   "/server"                2 minutes ago        Up 2 minutes                                   k8s_default-http-backend_default-http-backend-65dd5949d9-tnjmb_ingress-nginx_abfe4a3f-46b5-4e69-8d6c-989018a416c2_0
6dda29dc631b        rancher/coredns-coredns                           "/coredns -conf /etc…"   2 minutes ago        Up 2 minutes                                   k8s_coredns_coredns-6f85d5fb88-w6m59_kube-system_167eb98c-c4fc-4aa8-b2f1-bf17e3c112fe_0
05d35405d21d        rancher/cluster-proportional-autoscaler           "/cluster-proportion…"   2 minutes ago        Up 2 minutes                                   k8s_autoscaler_coredns-autoscaler-79599b9dc6-6k29k_kube-system_98fec6d6-31f4-4f01-9a6f-e9dc3b96ed92_0
5c7079401ee0        rancher/metrics-server                            "/metrics-server --k…"   2 minutes ago        Up 2 minutes                                   k8s_metrics-server_metrics-server-8449844bf-rrnf5_kube-system_33e60899-539e-4626-9b08-989c95dd4f17_0
2b064b6efccd        rancher/coreos-flannel                            "/opt/bin/flanneld -…"   2 minutes ago        Up 2 minutes                                   k8s_kube-flannel_canal-2qndj_kube-system_2f7059bc-e4f5-4533-88fe-98919e57ecd0_0
9c03082128f3        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_default-http-backend-65dd5949d9-tnjmb_ingress-nginx_abfe4a3f-46b5-4e69-8d6c-989018a416c2_2
6178be3c4a33        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_coredns-6f85d5fb88-w6m59_kube-system_167eb98c-c4fc-4aa8-b2f1-bf17e3c112fe_3
7590b6afb73b        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_coredns-autoscaler-79599b9dc6-6k29k_kube-system_98fec6d6-31f4-4f01-9a6f-e9dc3b96ed92_2
7f545f295f5d        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_metrics-server-8449844bf-rrnf5_kube-system_33e60899-539e-4626-9b08-989c95dd4f17_2
4ff3074950e4        rancher/calico-node                               "start_runit"            2 minutes ago        Up 2 minutes                                   k8s_calico-node_canal-2qndj_kube-system_2f7059bc-e4f5-4533-88fe-98919e57ecd0_0
3daecf4c0c52        rancher/calico-pod2daemon-flexvol                 "/usr/local/bin/flex…"   2 minutes ago        Exited (0) 2 minutes ago                       k8s_flexvol-driver_canal-2qndj_kube-system_2f7059bc-e4f5-4533-88fe-98919e57ecd0_0
6c0e2a96ac6a        rancher/nginx-ingress-controller                  "/usr/bin/dumb-init …"   2 minutes ago        Up 2 minutes                                   k8s_nginx-ingress-controller_nginx-ingress-controller-xb8bf_ingress-nginx_66817423-d3f2-457f-b752-d07032da04f2_0
cca347d46699        rancher/calico-cni                                "/opt/cni/bin/install"   2 minutes ago        Exited (0) 2 minutes ago                       k8s_install-cni_canal-2qndj_kube-system_2f7059bc-e4f5-4533-88fe-98919e57ecd0_0
77eab9495f0d        141c7125f5a2                                      "run.sh"                 2 minutes ago        Up 2 minutes                                   k8s_agent_cattle-node-agent-r6hwt_cattle-system_189a36b2-9aa4-4662-889f-93ebc40d8eb7_0
4379eb93d049        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_cattle-node-agent-r6hwt_cattle-system_189a36b2-9aa4-4662-889f-93ebc40d8eb7_0
7e2ce00572fa        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_nginx-ingress-controller-xb8bf_ingress-nginx_66817423-d3f2-457f-b752-d07032da04f2_0
75df8ad335f2        rancher/pause:3.2                                 "/pause"                 2 minutes ago        Up 2 minutes                                   k8s_POD_canal-2qndj_kube-system_2f7059bc-e4f5-4533-88fe-98919e57ecd0_0
8243950383a1        rancher/rke-tools:v0.1.65                         "nginx-proxy CP_HOST…"   2 minutes ago        Up 2 minutes                                   nginx-proxy
45330b1da716        rancher/hyperkube:v1.19.3-rancher1                "/opt/rke-tools/entr…"   2 minutes ago        Up 2 minutes                                   kubelet
860db1f96feb        rancher/hyperkube:v1.19.3-rancher1                "/opt/rke-tools/entr…"   2 minutes ago        Up 2 minutes                                   kube-proxy
546f8a48e274        rancher/rancher-agent:v2.5.1                      "run.sh -- share-roo…"   3 minutes ago        Exited (0) 3 minutes ago                       share-mnt
fceba56bb5c1        rancher/rke-tools:v0.1.65                         "/bin/bash"              3 minutes ago        Created                                        service-sidekick
```

@sevko21 does your cluster recovered from the error eventually? What was your cluster setup? Single node all roles, separate roles for each node? 

This was a Rancher server HA RKE `v1.19.2` 
Both the Downstream EC2 clusters were `v1.19.3`

**Comment by sevko21** [comment]
Hi  izaac,

I just upgraded to 2.5.2 and give another try to setup rke von aws nodes and got the same error. 
My Rancher server HA RKE: v1.19.3
EC2 cluster: v1.19.3

Unfortunately if I leave the error it does not recover. I tried different roles 1etcd 1 worker and 1 controlplane is the actual configuration. The cluster gets provisioned - after a while I got the message cluster provisioning complete and with the health check I got this error message. I also checked on the node:

telnet xxxx 6080
telnet: can't connect to remote host (127.0.1.1): Connection refused

Do you need more logs?

**Comment by sevko21** [comment]
Some more information. I was curious about the provisioning so I tried several other things:

Provisioned a server onpremise (RHEL 7.9). Started to build a rke with one node (role worker, etcd, controlplane)--> same issue
Provisioned rke with azure provider. Started to build a rke with 3 nodes (worker, etcd, controlplane)--> same issue

**Comment by superseb** [comment]
@sevko21 Can you please share the info on the HA cluster Rancher is running on? What version RKE is used and what k8s version, was this setup upgraded and if so any information on what version was first installed and upgraded to helps here. Also please provide if certificate configuration and/or server-url (where Rancher is accessible at) was ever changed.

**Comment by sevko21** [comment]
HA Cluster is running on 3 RHEL 7.9 machines with Kubernets Version v1.19.3
RKE version: v1.2.1
Started with a Rancher version at the beginning of the year (it was 2.4) and updated with each new version
Server-URL didn't change since then also certificate for the rancher deployment. I can check also if CA has changed

**Comment by superseb** [comment]
If its possible to provide access to the setup to investigate, please ping me on users slack via https://slack.rancher.io

**Comment by sevko21** [comment]
@superseb we can have a screen sharing on Monday. I'll contact you via slack

**Comment by mattmattox** [comment]
@superseb I ran into this same issue in the lab and the logs have been uploaded to the Box share.

**Comment by clausa** [comment]
Running into the same issue :-(
Is this related to https://github.com/rancher/rancher/issues/30325 ?

**Comment by luiscape** [comment]
We are also consistently experiencing this issue on v2.5.3. We attempt to create a cluster using the EC2 node driver and, with different combination of options from AMI, networking, and number of nodes / types, clusters fail to come-up in this exact step.

**Comment by superseb** [comment]
@clausa @luiscape Please add the information as requested in the issue, is this an upgraded setup, if yes, from what Rancher version did you upgrade and what initial version of Rancher was installed in this cluster?

**Comment by luiscape** [comment]
@superseb hey there, yes we were trying to upgrade from Rancher 2.4.8 to 2.5.3. We currently use a GCP load balanced in front of our Rancher HA cluster, which currently runs on 2 VMs on GCP.

**Comment by superseb** [comment]
@luiscape Anything you can share about the setup helps (type of  HA cluster, versions, backend type, instance types, LB config)

**Comment by superseb** [comment]
Everyone with this issue, please check if you have the `proxy` feature flag enabled in your setup. To check this, go to Global view, then open Settings, on top of that page click Feature Flags and disable `proxy`. Try provisioning a cluster again after disabling it (no restart or pod recycling is needed)

**Comment by sevko21** [comment]
I disabled the Feature Flag proxy today and deploye an rke cluster with amazon ec2 nodes. It looks fine so far:

Finished building up Kubernetes Cluster

**Comment by luiscape** [comment]
@superseb I confirm that the solution of turning off the experimental proxy has worked (we've now upgraded to 2.5.5). However, we've observed a long time waiting on

```
Waiting to register with Kubernetes
```

Maybe completely unrelated. Glad that we could now upgrade!

**Comment by superseb** [comment]
Waiting to register is either the core components being slow to start up (etcd/kube-apiserver) or the kubelet on that node being slow to register itself. The logs should have an indication what is going on. Else please file a new issue with steps to reproduce and logs included.

**Comment by superseb** [comment]
To test this issue:

- Set up Rancher v2.4.x
- Enable feature flag `proxy` (go to Global view, then open Settings, on top of that page click Feature Flags and enable proxy)
- Upgrade to v2.5.5
- Create a custom cluster and add nodes
- Verify that you see the error when agents are being deployed to the cluster
- Upgrade to v2.5.6-rc
- Create a custom cluster and add nodes
- Cluster should be built successfully (no error seen)

**Comment by izaac** [comment]
Reproduced as [described by Seb ](https://github.com/rancher/rancher/issues/29633#issuecomment-775793914)

```
[TRACE] clusterDeploy: deployAgent: applying agent YAML for cluster [c-zxkld], try #4: Error from server (InternalError): an error on the server ("dial tcp 127.0.0.1:6080: connect: connection refused") has prevented the request from succeeding
```

**Comment by izaac** [comment]
Rancher version: `v2.5.6-rc5` 
Rancher version: `master-head (02/22/2021)` _dbf18389c_

Custom clusters came up active, no errors as seen in `v2.5.5`

**Comment by bmoorthi** [comment]
> I get this with RKE too. Rancher v2.5.1 HA

have you got any solution
